modules/centrifuge/main.nf

```diff
@@ -0,0 +1,62 @@
+process CENTRIFUGE {
+    tag "$meta.id"
+    label 'process_high'
+
+    conda (params.enable_conda ? "bioconda::centrifuge=1.0.4_beta" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/centrifuge:1.0.4_beta--h9a82719_6' :
+        'quay.io/biocontainers/centrifuge:1.0.4_beta--h9a82719_6' }"
+
+    input:
+    tuple val(meta), path(reads)
+    path db
+    val save_unaligned
+    val save_aligned
+    val sam_format
+
+    output:
+    tuple val(meta), path('*report.txt')  , emit: report
+    tuple val(meta), path('*results.txt') , emit: results
+    tuple val(meta), path('*kreport.txt') , emit: kreport
+    tuple val(meta), path('*.sam')        , optional: true, emit: sam
+    tuple val(meta), path('*.fastq.gz')   , optional: true, emit: fastq
```

Comment:
I would rather export these as two separate channels - one for unmapped, one for mapped. I've had problems in other modules where it's been complicated to separate out them when I only wanted one.

Reply:
Sounds good! Would the following work?

```
tuple val(meta), path('*mapped.fastq.gz')       , optional: true, emit: fastq
tuple val(meta), path('*unmapped.fastq.gz')   , optional: true, emit: fastq
```

Reply:
No, would need separate emits: `emit: fastq_mapped` and `mit: fastq_unmapped` for example

Reply:
I updated the PR with the separate input channels and emits.